File: linkedlit/21_Merge_Two_Sorted_Lists.c
int MAX(int p, int q){
     return p>q?p:q;
 }
struct ListNode* mergeTwoLists(struct ListNode* l1, struct ListNode* l2) {
    if(!l1) return l2;
    if(!l2) return l1;
    struct ListNode *head = (struct ListNode*)malloc(sizeof(struct ListNode));
    struct ListNode *tmp = head;
    while(l1 && l2){
        if(l1->val <= l2->val){
            tmp->next = l1;
            l1 = l1->next;
        }else{
            tmp ->next = l2;
            l2 = l2->next;
        }
        tmp = tmp->next;
    } 
    if(l1){
        tmp->next = l1;
    }
    if(l2){
        tmp->next = l2;
    }
    return head ->next;
}

File: java/linkelist/21. Merge Two Sorted Lists.java
/**
 * Definition for singly-linked list.
 * public class ListNode {
 *     int val;
 *     ListNode next;
 *     ListNode(int x) { val = x; }
 * }
 */
class Solution {
    public ListNode mergeTwoLists(ListNode l1, ListNode l2) {
        ListNode l1node  = l1;
        ListNode l2node = l2;
        
        ListNode start = new ListNode(0);
        ListNode tmp = start;
        
        while (l1node !=null && l2node !=null) {
            if(l1node.val <= l2node.val) {
                tmp.next = l1node;
                l1node = l1node.next;
            } else {
                tmp.next = l2node;
                l2node = l2node.next;
            }
            tmp = tmp.next;
        }
        
        if (l1node != null) {
            tmp.next = l1node;
        }
        if (l2node != null) {
            tmp.next = l2node;
        }
        
        return start.next;
    }
}
